public/app/features/datasources/DataSourcesListItem.tsx

```diff
@@ -1,35 +1,151 @@
 import React, { PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import appEvents from 'app/core/app_events';
+import { deleteDataSource, loadDataSource, loadDataSources } from './state/actions';
+import { StoreState, CoreEvents } from 'app/types/';
 import { DataSourceSettings } from '@grafana/ui';
 
 export interface Props {
   dataSource: DataSourceSettings;
+  loadDataSources: typeof loadDataSources;
+  loadDataSource: typeof loadDataSource;
+  deleteDataSource: typeof deleteDataSource;
```

Comment:
if feels like it would be better to do this on the parent level and keep this Item component as simple

so instead of having these redux actions here, have a onDelete: () => void;  callback prop

Reply:
fixed